### PR TITLE
Adding lockable seed and delayed state bank.  Reorganizing validator structure

### PR DESCRIPTION
**Pull request by tavakyan**
This is incomplete so don't accept the review.  I wanted to just share the restructuring so you guys see where I'm trying to go with this.

### CODE REVIEW

**Comment by davesag** [comment]
@tavakyan looks like you need to set your editor defaults for Solidity to 4 spaces instead of 2.

**Comment by tavakyan** [comment]
The lockable seed is the minimal interface I'm hoping to have to add for the engine.  So basically create a seedable validator contract in rust to interface / get set this value. I'm hoping we figure out a way to to FTS on chain.  If not we'll have to add another interface that returns the coin indexes (getCoinIndices) to the epoch set in which case we won't even be using getValidators though we may be using report benign and malicious.

File: contracts/DelayedStakeBank.sol [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+pragma solidity ^0.4.23;
+
+import "./StakeBank.sol";
+
+
+contract DelayedStakeBank is StakeBank {
+    uint256 unstakeDelay;
+    mapping (address => uint256) lastStaked;
+
+    /// @param _token Token that can be staked.
```
Comment (davesag) [inline contracts/DelayedStakeBank.sol]: why three `///`?

can we stick to the usual

```
/**
 * block of comments
 */
```
style for block comments and
```
// single line comment
```
for single line comments?

File: contracts/EpochSet.sol [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import "./interfaces/LockableSeed.sol";
+import "./MajoritySet.sol";
+import "./PublicStakingSet.sol";
+
+
+contract EpochSet is ValidatorSet, LockableSeed {
+  PublicStakingSet publicSet;
+  MajoritySet privelagedSet;
+
+  // Return public and privelaged validators for this epoch
```
Comment (davesag) [inline contracts/EpochSet.sol]: `privileged` not `privelaged`

File: contracts/EpochSet.sol [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import "./interfaces/ValidatorSet.sol";
+import "./interfaces/LockableSeed.sol";
+import "./MajoritySet.sol";
+import "./PublicStakingSet.sol";
+
+
+contract EpochSet is ValidatorSet, LockableSeed {
+  PublicStakingSet publicSet;
+  MajoritySet privelagedSet;
```
Comment (davesag) [inline contracts/EpochSet.sol]: `privilegedSet` not `privelagedSet`

**Comment by tavakyan** [comment]
@davesag I think its because I copied and pasted the StakeBank stuff without linting

File: contracts/DelayedStakeBank.sol (hunk 1 above)
Comment (tavakyan) [reply contracts/DelayedStakeBank.sol]: Yes, I"ll change all the broken comments & documentation on next commit.

**Comment by tavakyan** [comment]
Or it could be we just return epochBalances from the epoch set.  That interface seems to make the most sense. @jaybutera @davesag any thoughts on this interface?

File: contracts/interfaces/LockableSeed.sol [hunk 4]
```diff
@@ -0,0 +1,16 @@
+pragma solidity ^0.4.23;
+
+import './Lockable.sol';
+
+
+contract LockableSeed is Lockable {
+  uint private seed;
```
Comment (davesag) [inline contracts/interfaces/LockableSeed.sol]: you could just make this `public` as the compiler will automatically create a getter function for every public state variable available but not a setter.  you can thus still restrict the `setSeed` to `onlyOwner` without issue.

**Comment by tavakyan** [comment]
I'm referring to create another contract interface called EpochBalances or something that has getEpochBalances() which returns [(Address, uint)].  Originally I was thinking we could return coinIndices, as a [address] and the node could call getBalance(address) on each of those to create the epoch balances.

File: contracts/interfaces/LockableSeed.sol (hunk 4 above)
Comment (tavakyan) [reply contracts/interfaces/LockableSeed.sol]: opps, I misssed that ty.

File: .eslintrc.json [hunk 5]
```diff
@@ -10,6 +10,7 @@
     "mocha": true
   },
   "rules": {
-    "prettier/prettier": ["error", { "singleQuote": true, "semi": false }]
+    "prettier/prettier": ["error", { "singleQuote": true, "semi": false }],
+    "no-unused-vars": ["error", { "argsIgnorePattern": "^__" }]
```
Comment (tavakyan) [inline .eslintrc.json]: @davesag can't tell if I'm doing something wrong.  I run "npm run lint:sol" and it doesn't seem to ignore this problem.  Same with truffle compile.

File: package.json [hunk 6]
```diff
@@ -15,26 +15,26 @@
...
+    "solidity-coverage": "^0.5.5"
   },
   "scripts": {
     "precommit": "lint-staged",
     "prettier": "prettier --single-quote --no-semi --write '**/*.{js,json}'",
     "eslint-check": "eslint --print-config .eslintrc.js | eslint-config-prettier-check",
     "test": "truffle test",
     "test:cov": "solidity-coverage",
-    "lint:sol": "solhint contracts/**/*.sol",
+    "lint:sol": "solhint contracts/**/**/*.sol",
```
Comment (tavakyan) [inline package.json]: npm run lint:sol wasn't working with all contracts until I added this. Not sure what is up.

File: .eslintrc.json (hunk 5 above)
Comment (tavakyan) [reply .eslintrc.json]: In the StakeBank contract there is a field in stakeFor and unstake called __data which is unused and is supposed to be ignored.

**Comment by tavakyan** [comment]
I will start adding unit tests once I get the burning and block rewards interface in place since right now its still being designed.

**Pull request merged**